### PR TITLE
[ROCm] FileCheck fixes

### PR DESCRIPTION
**Pull request by ekuznetsov139**
This PR patches several FileCheck pattern tests so that they pass on ROCm.

### CODE REVIEW

File: xla/service/gpu/fusions/input_slices_mlir_test.cc
Comment (akuegel) [inline xla/service/gpu/fusions/input_slices_mlir_test.cc:45]: I guess this is a matter of different order of sge and sle? Then it would be better to change to CHECK-DAG for those two lines.

File: xla/service/gpu/tests/reduce_atomic_min.hlo
Comment (akuegel) [inline xla/service/gpu/tests/reduce_atomic_min.hlo:42]: maybe separate lines CHECK-GCN and CHECK-PTX?

File: xla/service/gpu/tests/reduce_column_layout_change.hlo
Comment (akuegel) [inline xla/service/gpu/tests/reduce_column_layout_change.hlo:89]: maybe separate lines CHECK-GCN and CHECK-PTX?

File: xla/service/gpu/tests/reduce_column_layout_change.hlo
Comment (akuegel) [inline xla/service/gpu/tests/reduce_column_layout_change.hlo:91]: maybe separate lines CHECK-GCN and CHECK-PTX?

File: xla/service/gpu/tests/reduce_column_layout_change.hlo
Comment (akuegel) [inline xla/service/gpu/tests/reduce_column_layout_change.hlo:176]: maybe separate lines CHECK-GCN and CHECK-PTX?

Comment (ekuznetsov139) [reply xla/service/gpu/tests/reduce_column_layout_change.hlo:176]: if I do separate lines CHECK-GCN and CHECK-PTX for every load and store, it will mean a lot of code duplication

File: xla/service/gpu/tests/reduce_column_layout_change.hlo
Comment (ekuznetsov139) [reply xla/service/gpu/tests/reduce_column_layout_change.hlo:89]: There does not seem to be any value in checking that !5 (I think it's a metadata index). Might be better just to drop it entirely, and end the line at !llvm.loop. 

BTW this is one of the most common divergences between GCN and PTX in these tests, I've seen it in several other files.

File: xla/service/gpu/fusions/input_slices_mlir_test.cc
Comment (ekuznetsov139) [reply xla/service/gpu/fusions/input_slices_mlir_test.cc:45]: I'll try that

File: xla/service/gpu/tests/reduce_column_layout_change.hlo
Comment (akuegel) [reply xla/service/gpu/tests/reduce_column_layout_change.hlo:89]: Yes, ending the line early sounds also good. I would just like to avoid that the patterns become harder to read.

File: xla/service/gpu/fusions/input_slices_mlir_test.cc
Comment (ekuznetsov139) [reply xla/service/gpu/fusions/input_slices_mlir_test.cc]: Done.
Tangential question: is there some easy way to tell the compiler to limit to 512 threads for one of these tests? There is a llvm attribute amdgpu-flat-work-group-size but I can't figure out how to pass it to the test. In a few other tests, there are substantial differences between NV and AMD versions, because the compiler uses 512 for NV and 1024 for AMD.

**Review by akuegel — APPROVED** [review]

**Comment by akuegel** [comment]
Regarding your question about the 512 vs 1024 threads: I believe that is controlled here: https://github.com/openxla/xla/blob/main/xla/service/gpu/reduction_utils.cc#L86

**Comment by ekuznetsov139** [comment]
> Regarding your question about the 512 vs 1024 threads: I believe that is controlled here: https://github.com/openxla/xla/blob/main/xla/service/gpu/reduction_utils.cc#L86

After I asked it, I investigated further and discovered that the difference actually not caused by the number of threads, but by this logic here https://github.com/openxla/xla/blob/main/xla/service/gpu/fusions/reduction_base.cc#L91-L99

The setting for ROCm can probably be harmonized with CUDA, but I need to find a good way to benchmark the change.

**Comment by kamaljeeti** [comment]
Hi @ekuznetsov139 , any updates on this PR? Thanks

**Comment by ekuznetsov139** [comment]
> Hi @ekuznetsov139 , any updates on this PR? Thanks

@kamaljeeti I have addressed all feedback, I don't know why this PR is not being acted upon.

**Comment by kamaljeeti** [comment]
Hi @akuegel , can you look into this? Thanks.

**Comment by akuegel** [comment]
It seems the internal version of this PR was never approved, so it got garbage collected. I will approve again so that another internal version is created. The problem is that we always need at least 2 different people to approve, so if I approve here, I cannot approve the internal version.

**Review by akuegel — APPROVED** [review]

**Review by thomasjoerg — APPROVED** [review]

**Comment by ekuznetsov139** [comment]
There are conflicts now, I'm going to fix them, will the PR need to be approved again after that?

**Comment by thomasjoerg** [comment]
> There are conflicts now, I'm going to fix them, will the PR need to be approved again after that?

Not sure. Please reply to this thread once conflicts are resolved and I check and re-approve, if needed.

**Comment by ekuznetsov139** [comment]
I've resolved the conflicts, but now it fails XLA Linux CPU with a compiler error in a file that my PR does not touch.

**Review by akuegel — APPROVED** [review]

**Pull request closed without merging**
